File: server/tests/fixtures/alpha_rule_triggers/csharp/security_jwt_without_exp.cs
// Should trigger: security.jwt_without_exp
using System.IdentityModel.Tokens.Jwt;

public class JwtHandler {
    public string CreateToken(string subject) {
        var token = new JwtSecurityToken(
            issuer: "test",
            audience: "test"
            // No expiration!
        );
        return new JwtSecurityTokenHandler().WriteToken(token);
    }
}

File: server/tests/fixtures/alpha_rule_triggers/java/security_jwt_without_exp.java
// Should trigger: security.jwt_without_exp
import io.jsonwebtoken.Jwts;

public class JwtHandler {
    public String createToken(String subject) {
        // JWT without expiration claim!
        return Jwts.builder()
            .setSubject(subject)
            .signWith(key)
            .compact();
    }
}
